unity-environment/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -257,10 +257,9 @@ public abstract class Agent : MonoBehaviour
         void OnEnable()
         {
             textureArray = new Texture2D[agentParameters.agentCameras.Count];
-            for (int i = 0; i < brain.brainParameters.cameraResolutions.Length; i++)
+            for (int i = 0; i < agentParameters.agentCameras.Count; i++)
             {
-                textureArray[i] = new Texture2D(brain.brainParameters.cameraResolutions[i].width, 
-                    brain.brainParameters.cameraResolutions[i].height, TextureFormat.RGB24, false);
+                textureArray[i] = new Texture2D(1, 1, TextureFormat.RGB24, false);
```

Comment:
What is the reason behind this change?

Reply:
The agent can be initialized without a brain attached, in which case the old code would throw an error. In the new code we use the agent's camera list to initialize the texture2d array. But, since we don't know how big the textures should be, we initialize them to 1x1 to start with.

Reply:
Why do we want to allow the agent be initialized without a brain attached?

Reply:
We may want to programmatically attach brains during runtime, as is the case with `WallJump`.

Reply:
Oh I see.